common/djangoapps/xblock_django/user_service.py

```diff
@@ -81,7 +81,11 @@ def _convert_django_user_to_xblock_user(self, django_user):
             # need this check because profile is only created through registration flow
             full_name = full_name if full_name is not None and full_name != '' else "{} {}".format(
                 django_user.first_name, django_user.last_name)
+            first_name = django_user.first_name
+            # extract first name from full name, if profile does not have first name explicitly
+            first_name = first_name if first_name is not None and first_name != '' else full_name.split()[0]
```

Comment:
We can rely on `full_name` to extract `first_name` but we should check for a blank full name.
ps please `title()` the full name as well.

Reply:
`django_user.first_name` will return blank instead of `None`

Reply:
I have updated the code